### PR TITLE
Edit wallet: able to edit wallets and re-edit to original state

### PR DESCRIPTION
**Pull request by 0xdorifto**
Solution for https://github.com/iron-wallet/iron/issues/294

Before:

[Screencast from 17-08-2023 15:45:07.webm](https://github.com/iron-wallet/iron/assets/57015299/71b4cbf1-4c76-4974-b7e5-271cbe038d15)

After:

[Screencast from 17-08-2023 15:45:37.webm](https://github.com/iron-wallet/iron/assets/57015299/808ebc9e-9b60-4663-8427-c9fe27408c7f)

### CODE REVIEW

**Comment by naps62** [comment]
Hey. thanks for the help!

Before testing this: I think the issue maybe wasn't fully clear now that I re-read. the problem is general about all settings. The checkbox was just the one example I used

in general: editing settings, and then editing back to what the original state was disables the button.

I'll probably merge this anyway as a quick fix for the checkbox, which is the most direct way to reach the error, but the issue won't be fixed by this

**Comment by 0xdorifto** [comment]
@naps62 yeah, I thought it was something specific to the checkbox. I'll keep looking at it then, no need to merge this.

**Comment by 0xdorifto** [comment]
[simplescreenrecorder-2023-08-29_19.56.49.webm](https://github.com/iron-wallet/iron/assets/57015299/bd7f804a-3de7-4318-8a39-337df86428f2)
This is how it is looking, with the save button closing the accordion.

**Comment by 0xdorifto** [comment]
Does it look good to you @naps62?

File: gui/src/components/Settings/Wallet/JsonKeystore.tsx [hunk 1]
```diff
@@ -50,7 +53,8 @@ export function JsonKeystore({
           color="primary"
           variant="contained"
           type="submit"
-          disabled={!isDirty || !isValid}
+          disabled={(!isDirty || !isValid) && !isActuallyDirty}
+          onClick={() => setIsActuallyDirty(true)}
```
Comment (naps62) [inline gui/src/components/Settings/Wallet/JsonKeystore.tsx:57]: I'm not understanding this logic

clicking on the submit button makes the dirty state true? that sounds backwards. in fact, check this flow:

- edit an item (e.g.: the "dev account" checkbox)
- save the wallet, which closes the accordion
- open the wallet again. the save button is now enabled, even though it's in the original state

what I would imagine a correct solution does is:

- each wallet uses `defaultValues: wallet`. which is what's used to compare with current values and compute `isDirty`
- if we have this value updated (which should be happening automatically, but maybe there's some break along the stack I'm missing), this should "just work". Ideally we'd find out where the reactivity is missing and fix there

Comment (naps62) [reply gui/src/components/Settings/Wallet/JsonKeystore.tsx:57]: Thinking about this again, maybe its just a matter of calling https://react-hook-form.com/docs/useform/reset ?

**Comment by 0xdorifto** [comment]
@naps62 can you give me some feeback on this?

**Comment by naps62** [comment]
Removed the auto-close accordion part, since I'm having second thoughts about it (and because the current logic was a bit limiting in other ways).

Also had to tweak the types a bit, but otherwise looks good

**Pull request merged**